### ISSUE TITLE
Tags field accessibility

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#7326]
### Description
![Screenshot of a tags field.](https://user-images.githubusercontent.com/17884198/102951905-cc5cc080-4482-11eb-83a6-9f34c5bdea46.png)

- The x/remove icon has insufficient contrast.
- The x/remove icon is not keyboard-accessible.
- The x/remove icon should use a button element rather than an anchor element.
- Maybe a tag icon would be more appropriate than the plus icon?
- It would be helpful to use a list element for the added tags.
- The text input is missing a label.
- The typeahead behaviour. I think the pattern in [this demo](https://www.jqueryscript.net/demo/jQuery-Tags-Input-Plugin-with-Autocomplete-Support-Mab-Tag-Input/) makes more sense. **Related: #2921**
- Use aria live regions to announce tag suggestions and updates to tags.
- Not so accessibility-related, but I get a server error when I save a tags field with 2 of the same tags. Maybe this should just be silently corrected and saved?

### Steps to reproduce
1. Create and inspect a tags field.

### Additional info
- Craft version: 3.5.17.1

**Comment by brandonkelly** [comment]
Aspects of this are maybe not specific to Tags fields and should be fixed (e.g. the `x` button contrast), however anything here that is Tags field-specific should be placed at the bottom of the priority list, since Tags fields are going by the wayside in Craft 4.

**Comment by gcamacho079** [comment]
I'm closing this issue because it is outdated and no longer reflects the current version. 

The remove button contrast/keyboard issues were address in [PR #10169](https://github.com/craftcms/cms/pull/10169).
